sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2233,7 +2233,7 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
         .option("multiline", "true")
         .options(Map("encoding" -> "UTF-16BE"))
         .json(testFile(fileName))
-        .count()
+        .collect()
```

Comment:
just curious why going from count() to collect() here ?

Reply:
The test has to really touch JSON to detect encoding even without parsing. With this optimization `jackson` parser is not called at all in the case of `count()`. `collect()` guarantees that JSON parser will be invoked with wrong `encoding`.